# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: 52ac63af39a4
# Full Hash: 52ac63af39a4de5a2e8ea8d44071a3006f6a1b8d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-25 09:49:23
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 775fcd064844 -r 52ac63af39a4 docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Fri Apr 24 18:33:01 2020 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Fri Apr 24 18:33:04 2020 +0000
@@ -134,7 +134,7 @@
   void CanonicalDiscard();
 
   using Type = BrowsingContext::Type;
-  CanonicalBrowsingContext(BrowsingContext* aParent,
+  CanonicalBrowsingContext(WindowContext* aParentWindow,
                            BrowsingContextGroup* aGroup,
                            uint64_t aBrowsingContextId,
                            uint64_t aOwnerProcessId,